File: src/main/java/com/smpcore/command/SmpConfigCommand.java
package com.smpcore.command;

import com.smpcore.config.ConfigManager;
import com.smpcore.config.SmpSettings;
import com.smpcore.sidebar.SidebarManager;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;

import java.util.List;

public final class SmpConfigCommand implements CommandExecutor, TabCompleter {
    private final ConfigManager configManager;
    private final SidebarManager sidebarManager;

    public SmpConfigCommand(ConfigManager configManager, SidebarManager sidebarManager) {
        this.configManager = configManager;
        this.sidebarManager = sidebarManager;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!sender.hasPermission("smpcore.admin")) {
            sender.sendMessage(color("&cYou do not have permission to use this command."));
            return true;
        }

        if (args.length == 0) {
            sender.sendMessage(color("&eUsage: /" + label + " <reload|status>"));
            return true;
        }

        switch (args[0].toLowerCase()) {
            case "reload" -> {
                configManager.reload();
                sidebarManager.start();
                sender.sendMessage(color("&aSMPCORE configuration reloaded."));
                return true;
            }
            case "status" -> {
                SmpSettings settings = configManager.settings();
                for (String line : settings.statusText()) {
                    sender.sendMessage(color(line));
                }
                return true;
            }
            default -> {
                sender.sendMessage(color("&cUnknown subcommand. Use /" + label + " <reload|status>"));
                return true;
            }
        }
    }

    @Override
    public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
        if (args.length == 1) {
            return List.of("reload", "status").stream()
                    .filter(opt -> opt.startsWith(args[0].toLowerCase()))
                    .toList();
        }
        return List.of();
    }

    private String color(String text) {
        return ChatColor.translateAlternateColorCodes('&', text);
    }
}

File: src/main/kotlin/com/smpcore/command/SmpConfigCommand.kt
package com.smpcore.command

import com.smpcore.config.ConfigManager
import com.smpcore.sidebar.SidebarManager
import com.smpcore.util.Text
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.plugin.java.JavaPlugin

class SmpConfigCommand(
    private val plugin: JavaPlugin,
    private val configManager: ConfigManager,
    private val sidebarManager: SidebarManager
) : CommandExecutor, TabCompleter {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (!sender.hasPermission("smpcore.admin")) {
            sender.sendMessage(Text.c("&cYou do not have permission to use this command.")); return true
        }

        if (args.isEmpty()) {
            sender.sendMessage(Text.c("&eUsage: /$label <reload|status|set>")); return true
        }

        when (args[0].lowercase()) {
            "reload" -> {
                configManager.reload()
                sidebarManager.start()
                sender.sendMessage(Text.c("&aSMPCORE configuration reloaded."))
            }
            "status" -> configManager.settings().statusText().forEach { sender.sendMessage(Text.c(it)) }
            "set" -> {
                if (args.size < 3) {
                    sender.sendMessage(Text.c("&eUsage: /$label set <path> <value>")); return true
                }
                val path = args[1]
                val value = args.drop(2).joinToString(" ")
                val parsed = parseValue(value)
                plugin.config.set(path, parsed)
                plugin.saveConfig()
                configManager.reload()
                sidebarManager.start()
                sender.sendMessage(Text.c("&aUpdated config path &f$path&a = &f$parsed"))
            }
            else -> sender.sendMessage(Text.c("&cUnknown subcommand. Use /$label <reload|status|set>"))
        }
        return true
    }

    private fun parseValue(input: String): Any {
        if (input.equals("true", true) || input.equals("false", true)) return input.toBoolean()
        return input.toIntOrNull() ?: input.toDoubleOrNull() ?: input
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        alias: String,
        args: Array<out String>
    ): MutableList<String> {
        if (args.size == 1) {
            val options = listOf("reload", "status", "set")
            return options.filter { it.startsWith(args[0].lowercase()) }.toMutableList()
        }
        return mutableListOf()
    }
}
